### ISSUE TITLE
publish to peertube

### ISSUE THREAD

**Issue by albjeremias** [voc/voctopublish#83]
how hard would it be to implement this? It would be awesome because peertube supports webtorrent video streaming.. and also its part of the amazing #fediverse!

https://pypi.org/project/peertube-uploader/

**Comment by derpeter** [comment]
Implementing it as a releasing target should be simple.

But as far as i understand peertube someone would need to provide storage for that. Sadly all the distributed streaming implementation have the same issue, some one still needs to provide that storage.
So at least to use it in the voc context it would require more than 10TB that someone would need to provide. To make it distributed we talk about many times this amount. 
You would end up with running the same amount of server to provide the same service quality we currently have. 

Also you loose all privacy and make the internet a worse place by using torrent technology and gain basically nothing. The only benefit i see would be if you are in region where our CDN is not well reachable AND someone close to your shares the file you want to watch. But this sound statically unlikely to me.

TL;DR its not hard to implement but probably not worth the effort for the VOC. If you use the software in a different context where you don't have your own CDN this might be more useful.

**Comment by albjeremias** [comment]
I'm sorry, about privacy i can not agree with you.. because a CDN will not protect you from that.. or I guess what you meant with a VPN easily you can view the videos? 
About the 10TB, how are you solving this problem then? Just keep on using corporate CDNs? Aren't we locking ourselfs up with corps which decide too much and too detailed to censor our contents? Even them being not infringin any copyright law? And letting all this alt-right speech go on in their platforms? Fediverse and peertube are very interesting approaches, for this issues. And I don't think they are solving all the problems but we should find ways to support each other and understading limits and concerns about software :)

anyway i understand most of what you said, but I'd like to understand how could I solve that issue, just use youtube as my CDN? It's a current issue i'm trying to understand, how can i livestreaming without being DoS? Video is heavy on bandwidth.. How do you solve that? (I'm keeping on checking your repositories to understand it better, some stuff is not translated, but im glad a lot is.. i see a great effort.. 
 thanks a lot for the work its great tool)

**Comment by a-tze** [comment]
First of all, do you want this feature for the script, or for the content on media.ccc.de, or for you personal project? The answer depends heavily on that.

Second, you have a misconception. When someone says "CDN" in this thread, this does not automatically refer to some commercial provider. VOC has an independent CDN, meaning a bunch of rsyncing mirrors with a mirrorbits redirector as frontend. No censoring, no privacy violations. If you dont trust that, use TOR or similar.

Third, you cannot livestream without getting DoSed. Having more interested viewers than what your infrastructure can handle is a DoS situation, luckily you know before that it will happen. So you either have enough ressources (especially bandwidth) or you use some commercial platform or you have a problem. Thats about all the options there are. We try the first approach, having enough ressources, but this solution is by far not reproducable by everyone.

However, to be honest nothing of this all is actually related to this repo.

**Comment by derpeter** [comment]
@albjeremias atze summed it up already. I assume there is misunderstanding what this script does. It talks to our publishing ticket system and publishes the encoded videos to currently media.ccc.de (aka voctoweb) and youtube. Youtube is here only for the convenience of the user and to reach people who would not stumble on this content on a less visible platform than youtube.

So if you are planing to use this script and the ticket system to publish your content to peertube, we probably can assist you with that. For our own use case currently i don't see any benefit despite the theoretical one described above.

**Comment by derpeter** [comment]
@derchrisuk do you need this feature for your peertube experiments or can we close the ticket?

**Comment by derchrisuk** [comment]
As it will just "upload" the file back to PeerTube, I don't think I need this.
PeerTube has no way of using HTTP or local links for files.
